server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -103,7 +105,13 @@ public class SynonymsManagementAPIService {
         .build();
 
     public SynonymsManagementAPIService(Client client) {
+        this(client, MAX_SYNONYMS_SETS);
+    }
+
+    // Used for testing, so we don't need to test for MAX_SYNONYMS_SETS and put unnecessary memory pressure on the test cluster
+    SynonymsManagementAPIService(Client client, int maxSynonymsSets) {
```

Comment:
Instead of testing using a lot of synonyms, which made tests slow and consumed a lot of heap, I decided to artificially limit the max on tests